### PR TITLE
Updated README.md

### PR DESCRIPTION
**Pull request by AGaliciaMartinez**
Readme of the project. It includes the installation procedure, a description of the project and instructions for the benchmarks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,21 +1,107 @@
...
+
+    # The operation computed is <0|x*sz|0> = x <0|sz|0> = x
+    y = qutip.expect(x*sz, state)
+
+# dy/dx = 1
+tape.gradient(y, variable)  # 1
+```
+
+For a more involved example of how to use `GradientTape` for optimization
+purposes see the example notebook in `qutip_tensorflwo/examples`, which can be
```
Comment (AGaliciaMartinez) [inline README.md:70]:
```suggestion
purposes see the example notebook in `qutip_tensorflow/examples`, which can be
```

File: README.md [hunk 2]
```diff
@@ -1,21 +1,107 @@
...
+default a `Qobj` backed with either a `Dense` or `CSR` data container. To
+obtain a `Qobj` backed with a `TfTensor` it suffices to use the `to` method:
+```python
+sx = qutip.sigmax()  # Pauli X matrix
+sx.data  # Instance of `CSR`
+sx = sx.to('tftensor') # 'TfTensor' also works
+sx.data  # Instance of `TfTensor`
+```
+
+When importing qutip-tensorlow, operations are done using the default detected
```
Comment (AGaliciaMartinez) [inline README.md:45]:
```suggestion
When importing qutip-tensorflow, operations are done using the default detected
```

File: README.md [hunk 3]
```diff
@@ -1,21 +1,107 @@
...
+
+To use qutip-tensorflow you only need to include the import statement.
+```python
+import qutip_tensorflow
+```
+Once qutip-tensorflow is imported, it hooks into QuTiP adding a new data backed
+based on TensorFlow's Tensor. It is hence not necessary to use any of
+qutip-tensorflow's functions explicitly.
+
+The main class implemented in qutip-tensorflow is the `TfTensor` class that
```
Comment (AGaliciaMartinez) [inline README.md:20]:
```suggestion
The main class implemented in qutip-tensorflow is `TfTensor` that
```

File: README.md [hunk 4]
```diff
@@ -1,21 +1,107 @@
...
+development, so it is necessary to install it from source:
+```
+pip install git+https://github.com/qutip/qutip@dev.major
+```
+To install qutip-tensorflow from source:
+```
+pip install git+https://github.com/qutip/qutip-tensorflow
+```
 
 Support
```
Comment (AGaliciaMartinez) [inline README.md:93]: Add what is not supported yet.

File: README.md [hunk 5]
```diff
@@ -1,21 +1,107 @@
...
 -------
```
Comment (AGaliciaMartinez) [inline README.md:95]: Add how to use the benchmarks.

File: README.md [hunk 6]
```diff
@@ -1,7 +1,162 @@
...
+-`python benchmarks/benchmarks.py -k"add and -dense- and qutip_"`: runs only the
+benchmarks for `add` with dense random matrices for all the specialisations in
+QuTiP. 
+-`python benchmarks/benchmarks.py -k"expm and -512-"`: Runs only the
+benchmarks for `expm` for a matrix of size 512x512 (the size can only be
+2,4,8...,512,1024).
+-`python benchmarks/benchmarks.py -k"(tensorflow or numpy or qutip_dense) and
+-2-"`: Runs the benchmarks for every operation with hermitian
+matrices of size 2x2 represented with either `tensorflow`, `numpy` or the
+`qutip_dense` data type.
```
Comment (AGaliciaMartinez) [inline README.md:159]:
```suggestion
-`python benchmarks/benchmarks.py -k"test_linear_algebra" --collect-only`:
Shows all the available benchmarks. Useful to filter them with the `-k`
argument. 

-`python benchmarks/benchmarks.py -k"matmul"`: Runs only the benchmarks for
`matmul`.

-`python benchmarks/benchmarks.py -k"add and -dense-"`: Runs only the
benchmarks for `add` (addition) with dense random matrices. 

-`python benchmarks/benchmarks.py -k"add and -dense- and qutip_dense"`: runs only the
benchmarks for `add` with dense random matrices and only for the `qutip_dense`
data type. 

-`python benchmarks/benchmarks.py -k"add and -dense- and qutip_"`: runs only the
benchmarks for `add` with dense random matrices for all the specialisations in
QuTiP. 

-`python benchmarks/benchmarks.py -k"expm and -512-"`: Runs only the
benchmarks for `expm` for a matrix of size 512x512 (the size can only be
2,4,8...,512,1024).

-`python benchmarks/benchmarks.py -k"(tensorflow or numpy or qutip_dense) and
-2-"`: Runs the benchmarks for every operation with hermitian
matrices of size 2x2 represented with either `tensorflow`, `numpy` or the
`qutip_dense` data type.
```

**Review by quantshah — APPROVED** [review]

**Comment by quantshah** [comment]
Very well written README.

**Pull request merged**